jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-backend/src/test/java/org/jbpm/console/ng/cm/backend/server/RemoteCaseManagementServiceImplTest.java

```diff
@@ -326,4 +332,193 @@ private CaseMilestone createTestMilestone(String caseMilestoneId, String caseMil
+
+        List<CaseActionSummary> availableActions= testedService.getCaseActions(containerId,caseId,defaultSearchRequest,userId);
+
+        final ArgumentCaptor<List> captor = ArgumentCaptor.forClass(List.class);
+        verify(clientMock).getAdHocFragments(containerId,caseId);
+        verify(clientMock).findCaseTasksAssignedAsBusinessAdministrator(eq(caseId),eq(userId),captor.capture(),anyInt(),anyInt(),eq(sortStr),eq(sortAsc));
+        assertEquals("Ready", captor.getValue().get(0));
+
+        assertEquals(4,availableActions.size());
+        assertEquals(ts1_name,availableActions.get(0).getName());
+        assertEquals(ts2_name,availableActions.get(1).getName());
+        assertEquals(ahf_1_name,availableActions.get(2).getName());
+        assertEquals(ahf_2_name,availableActions.get(3).getName());
+    }
+
+    @Test
+    public void getCaseInProgressActionsTest() {
+        String sortStr="CreatedOn";
+        boolean sortAsc=true;
+        String userId = "userId";
+
+        CaseActionSearchRequest defaultSearchRequest = new CaseActionSearchRequest();
+        defaultSearchRequest.setSort(sortStr);
+        defaultSearchRequest.setSortOrder(sortAsc);
+        defaultSearchRequest.setFilterBy(CaseActionsFilterBy.IN_PROGRESS);
+
+        testedService.getCaseActions(containerId,caseId,defaultSearchRequest,userId);
+
+        final ArgumentCaptor<List> captor = ArgumentCaptor.forClass(List.class);
+        verify(clientMock,never()).getAdHocFragments(containerId,caseId);
+        verify(clientMock).findCaseTasksAssignedAsBusinessAdministrator(eq(caseId),eq(userId),captor.capture(),anyInt(),anyInt(),eq(sortStr),eq(sortAsc));
+        assertEquals("InProgress",captor.getValue().get(0));
+        assertEquals("Reserved",captor.getValue().get(1));
+    }
+
+    @Test
+    public void getCaseCompletedActionsTest() {
+        String sortStr="CreatedOn";
+        boolean sortAsc=true;
+        String userId = "userId";
+
+        CaseActionSearchRequest defaultSearchRequest = new CaseActionSearchRequest();
+        defaultSearchRequest.setSort(sortStr);
+        defaultSearchRequest.setSortOrder(sortAsc);
+        defaultSearchRequest.setFilterBy(CaseActionsFilterBy.COMPLETED);
+
+        testedService.getCaseActions(containerId,caseId,defaultSearchRequest,userId);
+
+        final ArgumentCaptor<List> captor = ArgumentCaptor.forClass(List.class);
+        verify(clientMock,never()).getAdHocFragments(containerId,caseId);
+        verify(clientMock).findCaseTasksAssignedAsBusinessAdministrator(eq(caseId),eq(userId),captor.capture(),anyInt(),anyInt(),eq(sortStr),eq(sortAsc));
+        assertEquals("Completed", captor.getValue().get(0));
+        assertEquals("Suspended", captor.getValue().get(1));
+        assertEquals("Failed", captor.getValue().get(2));
+        assertEquals("Error", captor.getValue().get(3));
+        assertEquals("Exited", captor.getValue().get(4));
+        assertEquals("Obsolete", captor.getValue().get(5));
```

Comment:
I see that this particular sequence of assertions is been repeating itself in this test class.
If the order of statuses in the captor list is not of importance in tests and I think it isn't, I suggest making less verbose assertion.
For example, create a list with these status values and then reduce the assertions to only one, by calling:
assertTrue(captorList.containsAll(ListOfStatusValues));